app/models/v2/report.rb

```diff
@@ -7,9 +7,45 @@ class Report < ApplicationRecord
     self.table_name = :reports
     self.primary_key = :id
 
+    SYSTEM_COUNT = lambda do
+      AN::NamedFunction.new(
+        'COUNT', [V2::System.arel_table[:id]]
+      ).filter(Pundit.policy_scope(User.current, V2::System).arel.ast.cores.first.wheres.first)
+    end
+
     # To prevent an autojoin with itself, there should not be an inverse relationship specified
     belongs_to :policy, class_name: 'V2::Policy', foreign_key: :id # rubocop:disable Rails/InverseOf
+    belongs_to :account, class_name: 'Account'
+
     has_many :tailorings, class_name: 'V2::Tailoring', through: :policy
-    has_many :systems, class_name: 'V2::System', through: :tailorings
+    has_many :test_results, class_name: 'V2::TestResult', dependent: nil, through: :tailorings
```

Comment:
You can link tailorings directly without the `through` as the `policy_id` in the tailoring is pointing to the report as well (foreign_key)